src/App.tsx

```diff
@@ -16,35 +17,98 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | ''>('');
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  enum SortType {
+    ALPHABET = 'alphabet',
+    LENGTH = 'length',
+    RESET = 'reset',
+  }
+
+  const visibleGoods = [...goodsFromServer].sort(
+    (good1: string, good2: string) => {
+      switch (sortField) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+        case SortType.RESET:
+          return 0;
```

Comment:
```suggestion
```
redundant.
and better to sort inside SortType.ALPHABET and SortType.LENGTH cases. it will avoid redundant iteration when no need to sort goods.

suggest to make function outside of compenent which will except goods from server, sortField and isVisible state and return visibleGoods. inside component just call this function.